nsqd/gossip_test.go

```diff
@@ -0,0 +1,128 @@
+package nsqd
+
+import (
+	"net"
+	"sort"
+	"sync"
+	"testing"
+	"time"
+)
+
+var gossipNotifyCh = make(chan struct{}, 20)
+var (
+	converging     = false
+	convergingLock = &sync.Mutex{}
+)
+
+func TestGossip(t *testing.T) {
+	var nsqds []*NSQD
+	var seedNode *NSQD
+	var tcpPorts []int
+
+	gossipNotify = func() {
+		convergingLock.Lock()
+		defer convergingLock.Unlock()
+		if converging {
+			gossipNotifyCh <- struct{}{}
```

Comment:
ok, I think you added this lock so that you could add logic around this channel send - if we made this channel send non-blocking, ie:

```go
select {
case gossipNotifyCh <- struct{}{}
default:
}
```

then we wouldn't need any of the locks... which would simplify all of this, particularly `converge()`